### DISCUSSION TITLE
fiveam-asdf not found

### DISCUSSION THREAD

**Discussion by samuelvimes32** [shop-planner/shop3#121] [Q&A] [answered]
Good morning,

I am new to using this planner and am trying to run some of the tests.lisp scripts to get to grips with the code.

Running tests.lisp from the blocks example I get the following error:
While evaluating the form starting at line 194, column 0
  of #P"C:/SHOP3 Planner/shop3-master/shop3/shop3.asd":
Unhandled MISSING-COMPONENT in thread #<SB-THREAD:THREAD "main thread" RUNNING {1005350143}>: Component "fiveam-asdf" not found

I have imported fiveam-asdf using quicklisp but this hasn't helped. Are you able to assist please?

Many thanks

Samuel

**Comment by rpgoldman** [comment, marked as answer]
@samuelvimes32 -- it would help a great deal if you would post a transcript of exactly what you did, from the start up of your lisp all the way through to the error.  Also, it would be helpful to know what lisp and OS you are using. 

To the best of my knowledge, you should be able to get all the dependencies (including fiveam-asdf) from quicklisp, but that's not how I load, so I don't know for sure.  And if you are loading with quicklisp, you will need to do `(ql:quickload "shop3")` not `(asdf:load-system "shop3")` (this is one of the reasons a transcript is critical).

In order for the tests to work, the SHOP3 repository is kept entirely self-contained, so if you wish you can find all of the dependencies in the `jenkins/ext/` subdirectory.  Note that these are *git submodules* so you would have to tell git to populate them with  `git submodule` subcommands.

**Reply by samuelvimes32** [reply]
Afternoon @rpgoldman, just got it working. Thanks for your help.

I am using windows OS and sbcl.

I hadn't realised that when downloading the content from github the linked data under Jenkins/Ext needed collecting manually.

**Reply by rpgoldman** [reply]
When you say "download," do you mean that you did some kind of downloading operation instead of doing a `git clone`? E.g., downloading a zip file?

I hadn't been thinking of people downloading SHOP3 instead of cloning it. Perhaps I should add a release download capability?  If that's right; will you please post an issue about that?
